src/main/java/chatlinitation/ChatViewmodel.java

```diff
@@ -0,0 +1,44 @@
+package chatlinitation;
+
+import entities.PrivateChat;
+
+/**
+ * chatviewmodel  connects our ChatView UI to our Entities.
+ * It provide data to the view, so that view can put that data on the screen.
+ * containing the message content and associated metadata
+ * this class contain privatechat, recipeint user and Textmessge.
+ */
+
+public class ChatViewmodel {
+
+    PrivateChat privatechat;
+    User recipient;
+    TextMessage txtmessage;
+
+
+ // constructor of the viewmodel
+    public ChatViewmodel() {
```

Comment:
I think you are missing the method signature

Reply:
maybe adding "ChatViewmodel(PrivateChat private chat, User recipient, TextMessage txtmessage )."?
However, I thought ViewModel class works like an adaptor connecting UI  to chat entities. .do we need it later?

Reply:
the lines directly below it are redundant if you dont include it.